ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsTask.java

```diff
@@ -140,7 +147,7 @@ public Object process(StatsAggregator statsAggregator) throws HiveException, Met
         StatsSetupConst.clearColumnStatsState(parameters);
       }
 
-      if (partfileStatus == null) {
+      if (partfileStatus == null && providedBasicStats == null) {
```

Comment:
note: these conditional seem to represent a different path - would be an early return an option?